lib/ResourceSet.cpp

```diff
@@ -18,37 +19,27 @@
 
 VCMI_LIB_NAMESPACE_BEGIN
 
-Res::ResourceSet::ResourceSet()
-{
-	resize(GameConstants::RESOURCE_QUANTITY, 0);
-}
-
 Res::ResourceSet::ResourceSet(const JsonNode & node)
 {
-	reserve(GameConstants::RESOURCE_QUANTITY);
-	for(const std::string & name : GameConstants::RESOURCE_NAMES)
-		push_back(static_cast<int>(node[name].Float()));
+	for(auto i = 0; i < GameConstants::RESOURCE_QUANTITY; i++)
+		at(i) = static_cast<int>(node[GameConstants::RESOURCE_NAMES[i]].Float());
 }
 
 Res::ResourceSet::ResourceSet(TResource wood, TResource mercury, TResource ore, TResource sulfur, TResource crystal,
 							TResource gems, TResource gold, TResource mithril)
 {
-	resize(GameConstants::RESOURCE_QUANTITY);
-	auto * d = data();
-	d[Res::WOOD] = wood;
-	d[Res::MERCURY] = mercury;
-	d[Res::ORE] = ore;
-	d[Res::SULFUR] = sulfur;
-	d[Res::CRYSTAL] = crystal;
-	d[Res::GEMS] = gems;
-	d[Res::GOLD] = gold;
-	d[Res::MITHRIL] = mithril;
+	this[Res::WOOD] = wood;
+	this[Res::MERCURY] = mercury;
+	this[Res::ORE] = ore;
+	this[Res::SULFUR] = sulfur;
+	this[Res::CRYSTAL] = crystal;
+	this[Res::GEMS] = gems;
+	this[Res::GOLD] = gold;
+	this[Res::MITHRIL] = mithril;

```

Comment:
Not sure whether we need to keep mithril. If/when it eventually gets used it'll likely be in a different way.
Might lead to some side effects though...

Reply:
About mithril removal - maybe it will break wog maps? I do not think than mithril placeholder is bad, I vote for keep it, at least for avoid side effects.

Reply:
Well, we shouldn't make implementing it any more difficult than ii is now.

In fact with current configurable objects it should be already possible to give and take Mithril.

Reply:
at removed.

Reply:
Agree with @DjWarmonger